developing-without-a-build/article.md

```diff
@@ -0,0 +1,257 @@
+
+### Caching
+Because we aren't bundling all of our code into just a few files, we don't have to set up any elaborate caching strategies. Your web server can use the file system's last modified timestamp to return a 304 if the file hasn't changed.
+
+You can test this in your browser by turning off `Disable cache` and refreshing:
+
+![](./images/caching.png)
+
+## Non-js modules
+So far we've only looked into javascript modules, and the story is looking pretty complete. It looks like we've got the tools needed to build different kinds of applications. But on the web we're not just writing javascript, we need to deal with other langauges as well.
+
+The good news is that there are proposals for html, css and json modules and all major browser vendors seem to be supportive of them:
+- [json modules](https://github.com/whatwg/html/issues/4315)
+- [html modules](https://github.com/w3c/webcomponents/issues/645)
+- [css modules](https://github.com/w3c/webcomponents/issues/759)
+
+The bad news is that they're not available just yet, and it will be a while for all browsers to support them. We have to look for some solutions in the meantime.
+
+### JSON
+In Node JS it's possible to import json files from javascript. These becomes available as javascript object. In web projects this is used frequently as well. There are many build tool plugins to make this possible.
+
+Until browsers support json modules, we can either just use a javascript module or use fetch to retreive the json files. See the `import.meta.url` section of an example of using fetch.
+
+### HTML
+Over time web frameworks have solved html templating in different ways, for example by placing HTML inside javascript strings. Lately JSX has become a popular format for embedding HTML inside javascript, but it is not going to run natively in the browser without some kind of transformation.
+
+With es2015/es6 we can use tagged template string literals to embed html inside js, and use it for doing efficient DOM updates. This runs natively in the browser and still offers a great developer experience. There are some really good production ready and feature complete libraries that can be used for this:
+
+- [htm, JSX using template literals. Works with libraries that use JSX, such as react](https://www.npmjs.com/package/htm)
+- [lit-html, a html templating library](https://www.npmjs.com/package/lit-html)
+- [lit-element, integrates lit-html with web components](https://www.npmjs.com/package/lit-element)
+- [haunted, a functional web components library with react-like hooks](https://www.npmjs.com/package/haunted)
+- [hybrids, another functional web component library](https://www.npmjs.com/package/hybrids)
+- [hyperHTML, a html templating library](https://www.npmjs.com/package/hyperhtml)
+(TODO: more libraries?)
```

Comment:
Anyone got suggestions for this?

Reply:
show an example of loading html as a string with `import(htmlFile).then(handleAsText)`

Reply:
Im not sure, is that really realistic and feasible?

Reply:
it's how `<html-include>` works https://github.com/justinfagnani/html-include-element/blob/b06b172348b8deecd71df3d0015e0f1f31023509/html-include-element.js#L74

Reply:
* meant `fetch`

Reply:
Added a section on this